include/deal.II/grid/grid_tools.h

```diff
@@ -230,6 +230,43 @@ namespace GridTools
   double
   cell_measure(const T &, ...);
 
+  /**
+   * Computes an aspect ratio measure for all cells and fills a vector with one
+   * entry per cell, given a @p triangulation and @p mapping. The aspect ratio
+   * of a cell is defined as the ratio of the maximum to minimum singular value
+   * of the Jacobian, taking the maximum over all quadrature points of a
+   * quadrature rule specified via @p quadrature. For example, for the special
+   * case of rectangular elements in 2d with dimensions $a$ and $b$ ($a \geq
+   * b$), this function returns the usual aspect ratio definition $a/b$. The
+   * above definition using singular values is a generalization to arbitrarily
+   * deformed elements. This function is intended to be used for $d=2,3$ space
+   * dimensions, but it can also be used for $d=1$ returning a value of 1.
+   *
+   * @note Inverted elements do not throw an exception. Instead, a value of inf
+   * is written into the vector in case of inverted elements.
+   *
+   * @note Make sure to use enough quadrature points for a precise calculation
+   * of the aspect ratio in case of deformed elements.
+   *
+   * @author Niklas Fehn, Martin Kronbichler, 2019
+   */
+  template <int dim>
+  Vector<double>
```

Comment:
can you clarify the size of the return value? I assume this is of length n_active_cells()? What happens with ghost or artificial cells?

Reply:
I don't see why the size of the vector should be different than what you have written. Can you please share your concerns? Do you want to assert the size?

Reply:
It seems (active) ghost and artificial cells have an aspect ratio of 0.0. Is that what we want?

Reply:
Sounds as `inf` to me. We should document that in any way.

Reply:
if you use inf for ghost/artificial then taking the max over the vector is no longer a sensible operation. :-)
So either denote 0.0 for cells where no information is available (ghost, artificial), or do not include them in the return value.

@nfehn, my comment here refers to the fact that you have to document the size of the return value. You have two options:
A) size is n_active_cells() and you need to decide what value is set for ghost and artificial cells.
B) size is the number of locally owned cells. (we currently don't even have a function that returns this number).

Reply:
@tjhei do you mean `parallel::TriangulationBase::n_locally_owned_active_cells`?  We could move that to `Triangulation` and override it in `parallel::TriangulationBase`...

Reply:
What meaning does ``locally_owned`` have for a non-parallel ``Triangulation``? But I agree that some basic deal.II functionality should be extended. From an OO design perspective, it should not be necessary to discuss ghost and artificial cells in this context at all.

Reply:
Improved documentation regarding vector size in https://github.com/dealii/dealii/pull/8894/commits/28a740482db88d0cd605140b889e860f8e001730 .

I intentionally do not mention ghost or artificial cells (assume someone introduces a third type of non locally-owned cells and the inconsistencies this would introduce, see Open-Closed Principle).

Reply:
To be consistent with output writers, we need to have the size of the vector `n_active_cells()` rather than the number of locally owned active cells and skip non-owned cells.

Reply:
Please add a sentence like
"@note In parallel computations the return value will have the length n_active_cells but the aspect ratio is only computed for the cells that are locally owned and placed at index CellAccessor::active_cell_index(), respectively. All other values are set to 0."

Reply:
done in https://github.com/dealii/dealii/pull/8894/commits/da7e69142eb66d977e7c02e0959c31b0163d70c1